tutoriale/Get Started with Pytest.md

```diff
@@ -0,0 +1,220 @@
+# Get Started with Pytest
+
+Pytest jest framework-iem, który ułatwia pisanie testów jednostkowych w Pythonie. Pomaga to osiągać *minimalnym* kodem. Jest również świetnym narzędziem do rozpoczęcia nauki testowania funkcji.
+
+## Intro
+
+"*Test jednostkowy – metoda testowania tworzonego oprogramowania poprzez wykonywanie testów weryfikujących poprawność działania pojedynczych elementów programu – np. metod lub obiektów w programowaniu obiektowym lub procedur w programowaniu proceduralnym.*" [Wikipedia](https://pl.wikipedia.org/wiki/Test_jednostkowy)
+
+Na wstępie należy podkreślić dwa fakty:
+1. Testy powstały z myślą o **aplikacjach**, co oznacza, że mają one testować **metody** z danych **klas** np. czy zwracają odpowiedni format, wymiar, czy nie ma nulli itp. Jest to najwygodniejsza forma to rozbudowanych aplikacji.
+
+> Z uwagi na charakter naszego zadania oraz strukturę projektu, testować będziemy wyłącznie elementy z `/lib`, a w tym te użyte potem w notatnikach.
+
+2. W programowaniu TDD (Test-driven development), wychodzi się od **wpierw** napisania testów, a następnie napisania aplikacji.
+
+> Z uwagi na nierównomiene rozłożenie umiejętności w programowaniu TDD wewnątrz zespołu, ta metodyka nie będzie wymuszana podczas tworzenia aplikacji.
+
+# Aplikacja
+
+## Struktura
+
+Standardowa struktura projektu (z reguły) zawiera folder */tests*. Służy on przechowywaniu funkcji testujących (testów jednostkowych). Natomiast aplikacja będzie przechowywana w innym folderze np. */lib*, */app*, */model*, */predictor* itp.
+
+Utwórzmy strukturę naszej aplikacji:
+```
+PyTest:
+    lib:
+        __init__.py
+        bmi_calc.py
+        bmi_calc.ipynb
+    tests:
+        __init__.py
+        input_test.py
+```
+
+> Uwaga! Pytest **odszukuje** testy jednostkowe po nazwach plików. Jeżeli wykryje plik rozpoczynający się od "test" lub kończący się na "test", tj. test_* lub *_test, wtedy **domyślnie** uzna ten plik za test jednostkowy i będzie z niego korzystać. Dodatkowo, należy pamiętać, aby **testy jednostkowe (funkcje) rozpoczynały się od `test_*`**!
+
+## Uruchamianie
+
+Jeżeli chcemy uruchomić nasze testy:
+- w przypadku obiektowym - wywołujemy komendę `pytest` w folderze nadrzędnym (w naszym przypadku /bmi_project), poprzez wykorzystanie terminala
+
+![img](pytest_files/pytest1.png)
```

Comment:
To są linki do obrazków - ale nie widzę tych plików w naszym repo.
Jak kraść - to brać całe kolekcje :-)